comp/process/agent/agentimpl/agent.go

```diff
@@ -88,19 +92,32 @@ func newProcessAgent(p processAgentParams) optional.Option[agent.Component] {
 }
 
 func (p processAgent) start(ctx context.Context) error {
-	p.Log.Info("process-agent starting")
+	p.Log.Debug("Starting the process-agent component")
 
 	chks := make([]string, 0, len(p.Checks))
 	for _, check := range p.Checks {
 		chks = append(chks, check.Name())
 	}
-	p.Log.Info("process-agent checks", log.Object("checks", chks))
+	p.Log.Debug("process-agent checks", log.Object("checks", chks))
 
+	// start the submitter
+	if err := p.Submitter.Start(); err != nil {
+		return err
+	}
```

Comment:
Why does the process agent handles the lifecycle of the submitter ? What happens if another component tries to use the submitter before the process agent calls `Start` ? I think this is the kind of shared state that are prone to create race conditions that we're trying to remove from the Agent.

Component should not control other comp lifecycle as FX handles it for us.

Reply:
The below is an abbreviated form of the dependency graph for the process-agent component.
   `HostInfo -> Runner -> Optional[Process-agent]`
   `Submitter ->  Optional[Process-agent]`

So the process-agent component has the config checks to determine whether to run. As those are dependencies of the process-agent component they have to be created first. When we have the process agent component disabled these components will be created and their on start hooks will be called and they will run but provide no functionality. 

Do you have any suggestions on how to control these types of optional dependencies? One approach I was thinking of was to push the config check into the dependency and return an Optional. When it's disabled it would return None and the Start/Stop hooks would not run. I'm not sure if that's the right approach.

Reply:
In previous instances, we created a parameter dependency that would have a simple bool for whether dependent components should disable themselves: for example Secrets does this https://github.com/DataDog/datadog-agent/blob/acd202029ca9173e256e69eb2ae2b8a2dd88a556/comp/core/secrets/params.go#L10. Would something like that work as a way to control how the Lifecycle hooks behave?